acceptancetests/assess_persistent_storage.py

```diff
@@ -27,6 +33,116 @@
 __metaclass__ = type
 log = logging.getLogger("assess_persistent_storage")
 
+def assess_charm_deploy_and_remove_single_storage(client):
+    """This function tests charm deployment and removal with single
+       persistent storage unit.
+
+       Steps taken to the test:
+       - Deploy dummy-storage charm with a single block storage unit (ebs)
+         and a single filesystem storage unit (rootfs).
+       - Check charm status once the deployment is done.
+       - Check charm storage list once the deployment is done.
+       - Remove the deployed charm, the output should indicate that the
+         rootfs storage unit will be removed, ebs storage unit will be
+         detached.
+       - Check storage list again to ensure ebs storage units really exists.
+
+       :param client: ModelClient object to deploy the charm on.
+    """
+
+    charm_name = 'dummy-storage'
+    charm_path = local_charm_path(
+        charm=charm_name, juju_ver=client.version)
+    log.info(
+        '{} is going to be deployed with 1 ebs block storage unit and '\
+        '1 rootfs filesystem storage unit.'.format(charm_name))
+    log.info(
+        'juju deploy ${JUJU_REPOSITORY}/charms/dummy-storage '\
+        '--storage single-blk=ebs --storage single-fs=rootfs')
+    # client.deploy() doesn't support repeat parameters,
+    # although juju deploy supports it
+    #client.deploy(
+    #    charm_path,
+    #    storage='single-blk=ebs',
+    #    storage='single-fs=rootfs')
+    client.get_juju_output(
+        'deploy', '/home/noname/work/Canonical/workbench/repository/charms/dummy-storage',
+        '--storage', 'single-blk=ebs',
+        '--storage', 'single-fs=rootfs',
+        include_e=False)
+    client.wait_for_started()
+    client.wait_for_workloads()
+
+    log.info('juju status --format json')
+    status_output = json.loads(
```

Comment:
In get_status(), until_timeout() is involved, which is the issue I mentioned before - it constantly execute the statement inside it, and if the command is "heavy-weight", error may happen. There was an example when I was trying to get storage list while using until_timeout() to run juju storage, items were missing from the result, but worked fine by using sleep(). In this case it will use juju status, even more heavier than juju storage.